github/MainClass.py

```diff
@@ -97,17 +83,26 @@ def __init__(
         login_or_token=None,
         password=None,
         jwt=None,
-        base_url=DEFAULT_BASE_URL,
-        timeout=DEFAULT_TIMEOUT,
+        app_id=None,
+        app_private_key=None,
+        app_installation_id=None,
+        app_token_permissions=None,
```

Comment:
Looks like these four go together in many places in this PR. What do you think about moving them into a class so we are carrying around only a single argument, e.g. `app_auth`.

Ideally, the now four existing ways for authentication (token, login and password, jwt, app auth) should be a single argument `auth` of a single base class (e.g. `GithubAuth`) with different implementations (`GitHubTokenAuth`, `GitHubLoginAuth`, `GitHubJwtAuth`, `GitHubAppAuth`). That would simplify a lot of signatures and stop growing signatures when new auths are added. But that is out-of-scope of this PR.

Reply:
That is a good point and it's done :)